### PR TITLE
GSW-872 detailed condition to check staking reward

### PR DESCRIPTION
**Pull request by r3v4s**
# New
1. more condition to calculate staking reward https://github.com/gnoswap-labs/gnoswap/commit/128dd4f7535d3ee8432154369791aebff8d35431

# Chore
2. better comments :D

# Fix
3. bunch of tc

### CODE REVIEW

File: staker/_TEST_staker_collect_reward_test.gno [hunk 1]
```diff
@@ -158,8 +158,8 @@ func TestUnstakeToken(t *testing.T) {
 		shouldEQ(t, gnft.OwnerOf(tid(2)), test1)
 
 		// check reward
-		shouldEQ(t, gns.BalanceOf(a2u(test1)), 100000000325)    // internal
```
Comment (mconcat) [inline staker/_TEST_staker_collect_reward_test.gno:161]: Why is this number has been changed?

File: staker/reward_math.gno [hunk 2]
```diff
@@ -105,7 +110,7 @@ func rewardMathComputeExternalRewardAmount(tokenId uint64, deposit Deposit, ince
 		monthlyReward = incentive.rewardAmount / 3
 	case incentiveDuration > TIMESTAMP_90DAYS:
 		// 1 second reward == total reward amount / reward duration
-		monthlyReward = incentive.rewardAmount / bigint(incentiveDuration)
+		monthlyReward = incentive.rewardAmount / bigint(incentiveDuration) * 2592000 // 30 days
```
Comment (mconcat) [inline staker/reward_math.gno:113]: Was this previously a bug?

Comment (r3v4s) [reply staker/reward_math.gno:113]: yes it was

File: staker/_TEST_staker_collect_reward_test.gno (hunk 1 above)
Comment (r3v4s) [reply staker/_TEST_staker_collect_reward_test.gno:161]: due to bit of extra function call in new test case, staking duration is changed and therefore balance is litte differ

File: _setup/gns/gns.gno [hunk 3]
```diff
@@ -15,8 +15,8 @@ var (
 )
 
 const (
-	admin                   std.Address = "g12l9splsyngcgefrwa52x5a7scc29e9v086m6p4" // r3v4_xxx: CHANGE WHEN DEPLOYING TO OFFICIAL NETWORK
-	INTERNAL_REWARD_ACCOUNT std.Address = "g1paqttvcjcluuya9n9twyw7yacv54mt7ld3gvzm" // r3v4_xxx: CHANGE WHEN DEPLOYING TO OFFICIAL NETWORK
+	admin                   std.Address = "g12l9splsyngcgefrwa52x5a7scc29e9v086m6p4" // GSA, r3v4_xxx: CHANGE WHEN DEPLOYING TO OFFICIAL NETWORK
```
Comment (notJoon) [inline _setup/gns/gns.gno:18]: What is the meaning of `GSA` and `IRA` here?

**Review by notJoon — APPROVED** [review]

**Review by mconcat — APPROVED** [review]

**Pull request merged**